docs/aws/audit/kubernetesmonitoring/rules/ecs_task_definition_user_for_host_mode.mdx

```diff
@@ -23,6 +23,273 @@ CBP
+
+Remember, tasks running in host network mode have the same networking permissions as the host itself, so it's important to limit the permissions of these tasks to only what they need to function.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can download it from the official AWS website and configure it using the "aws configure" command. You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. Once the AWS CLI is set up, you can list all the ECS tasks using the following command:
+
+   ```
+   aws ecs list-tasks --cluster <cluster-name>
+   ```
+
+   Replace `<cluster-name>` with the name of your ECS cluster. This command will return a list of task ARNs.
+
+3. For each task ARN, describe the task to get its network mode. Use the following command:
+
+   ```
+   aws ecs describe-tasks --cluster <cluster-name> --tasks <task-arn>
+   ```
+
+   Replace `<cluster-name>` with the name of your ECS cluster and `<task-arn>` with the ARN of the task. This command will return a JSON object that includes the task's network mode.
+
+4. Check the "networkMode" field in the returned JSON. If it is set to "host", then the task is using host network mode. This is a potential misconfiguration as tasks with host network mode have the same network access as the host, which can lead to security issues if the task is compromised.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary libraries in Python. You will need boto3 for AWS, azure-mgmt-resource for Azure, and google-cloud-resource-manager for GCP. 
+
+```python
+import boto3
+from azure.mgmt.resource import ResourceManagementClient
+from google.cloud import resource_manager
+```
+
+2. For AWS, create a session using your credentials and initialize the ECS client. Then, list all the task definitions and check their network mode. If it's 'host', check the permissions.
+
+```python
+session = boto3.Session(aws_access_key_id='YOUR_ACCESS_KEY',
+                        aws_secret_access_key='YOUR_SECRET_KEY',
+                        region_name='us-west-2')
+ecs_client = session.client('ecs')
+
+def check_task_permissions():
+    response = ecs_client.list_task_definitions()
+    for task in response['taskDefinitionArns']:
+        task_def = ecs_client.describe_task_definition(taskDefinition=task)
+        if task_def['taskDefinition']['networkMode'] == 'host':
+            # Check permissions here
+```
+
+3. For Azure, you need to authenticate and initialize the Kubernetes client. Then, list all the pods and check their network policy. If it's 'hostNetwork', check the permissions.
+
+```python
+credentials = ServicePrincipalCredentials(
+    client_id='YOUR_CLIENT_ID',
+    secret='YOUR_SECRET',
+    tenant='YOUR_TENANT_ID'
+)
+kubernetes_client = KubernetesClient(credentials, 'YOUR_SUBSCRIPTION_ID')
+
+def check_pod_permissions():
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>